app/src/lib/stores/app-store.ts

```diff
@@ -1780,6 +1780,40 @@ export class AppStore extends TypedBaseStore<IAppState> {
     this._updateCurrentPullRequest(repository)
     this.updateMenuItemLabels(repository)
     this._initializeCompare(repository)
+    this.refreshRepositoryState([repository])
```

Comment:
@vanessayuenn I've clarified the signature here to help make it's intent clearer, hopefully this makes our discussions about performance better. I'm still not sold on the `refreshRepositoryState` name as it feels too generic, maybe `refreshRepositoryContext` here but even that doesn't feel clearer...